pkg/apis/operator/v1alpha1/validation/validation.go

```diff
@@ -78,6 +79,24 @@ func ValidateGardenUpdate(oldGarden, newGarden *operatorv1alpha1.Garden) field.E
 	return allErrs
 }
 
+func validateRuntimeCluster(runtimeCluster operatorv1alpha1.RuntimeCluster, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if cidrvalidation.NetworksIntersect(runtimeCluster.Networking.Pods, runtimeCluster.Networking.Services) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networking", "services"), runtimeCluster.Networking.Services, "pod network of runtime cluster intersects with service network of runtime cluster"))
+	}
+	if runtimeCluster.Networking.Nodes != nil {
+		if cidrvalidation.NetworksIntersect(*runtimeCluster.Networking.Nodes, runtimeCluster.Networking.Pods) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("networking", "nodes"), *runtimeCluster.Networking.Nodes, "node network of runtime cluster intersects with pod network of runtime cluster"))
+		}
+		if cidrvalidation.NetworksIntersect(*runtimeCluster.Networking.Nodes, runtimeCluster.Networking.Services) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("networking", "nodes"), *runtimeCluster.Networking.Nodes, "node network of runtime cluster intersects with service network of runtime cluster"))
+		}
```

Comment:
Why do we need this?
I am completely in favour of ensuring network disjointness where required. However, I am not sure I understand what the service IPs in the virtual cluster are used for. There are certainly no pods backing the services. Hence, the IPs a purely virtual. If that is the case they could very well overlap with any of the given ranges.
Do you have some future scenario in mind?

Reply:
You're right, perhaps they could overlap. OTOH, isn't it better to be on the safe side and enforce disjointedness, just like we do for `Shoot`s? Who knows which vulnerability comes up in the future. If runtime and virtual cluster CIDRs are the same, services in the runtime cluster could perhaps be accessed which should not be allowed. We still have additional protection via `NetworkPolicy`s, but still. However, I agree that all this is rather hypothetical. If you insist, I can drop this check, of course.

Reply:
The above IP ranges will be used in networkPolicy here https://github.com/gardener/gardener/blob/0885a29a2443b3a5d4846b62d3d879182dac3014/pkg/controller/networkpolicy/reconciler.go#L341

What I understood is this netpol will allow egress traffic from the pods, nodes, service CIDRs provided above, In that case, shouldn't the `.Networking.Pods` be an optional field and not be set when not needed?

Reply:
As @ScheererJ has mentioned above that -
> There are certainly no pods backing the services.

Why do we need the PodIPs for the virtual cluster in the first place? I was suggesting to make PodIPs field optional in `RuntimeNetworkConfig`.

Reply:
I still can't follow, sorry. Where do you see that we "need the PodIPs for the virtual cluster"? We only need them for the runtime cluster to properly define the standard `NetworkPolicy`s like `allow-to-private-networks`. Any pod running in the runtime cluster could use it which allows access to private networks except for access to other pods in the runtime cluster.

Reply:
Ok, Thanks. I got confused with the first comment and the code selection for it.